spec/deaf_grandma_spec.rb

```diff
@@ -9,10 +9,22 @@
   end
   
   it "says 'NOT SINCE 1964!' when we yell" do
-    #implement your test here
+    expect(script.speak("YO GRANDMA!")).to eq "NOT SINCE 1964!"
   end
 
-  it "EXTRA CREDIT: How would you test yelling BYE?" do
-    #implement your test here
+  bye_examples = {
+    'BYE 3 times in a row' => (1..3).collect { "BYE" },
+    'BYE, then speak regularly, and then yell BYE twice' => [
+      "BYE", "It's been fun!", "BYE", "BYE"
+    ],
+    'BYE, then speak Bye, then yell BYE twice more' => [
+      "BYE", "Bye", "BYE", "BYE"
+    ] 
+  }
+  bye_examples.each_pair do |spec, phrases|
+    it "says 'SEE YOU LATER SONNY!' when we yell #{spec}" do
+      result = phrases.inject(nil) { |r, phrase| script.speak(phrase) }
```

Comment:
nice use of `inject(nil)` here. you may want to take a look at http://betterspecs.org/ but other than that this is great!